src/OSPSuite.Core/Services/ResidualsVsTimeChartService.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using OSPSuite.Core.Domain;
+using System.Collections.Generic;
+using OSPSuite.Core.Chart;
+using OSPSuite.Core.Domain.Data;
+using OSPSuite.Core.Domain.ParameterIdentifications;
+using OSPSuite.Core.Domain.UnitSystem;
+using OSPSuite.Core.Extensions;
+
+namespace OSPSuite.Core.Services
+{
+   public interface IResidualsVsTimeChartService
+   {
+      /// <summary>
+      ///    This will be the name of the curve added to the chart representing the Zero Curve
+      /// </summary>
+      string Zero { get; }
+
+      /// <summary>
+      ///    Adds the zero marker curve to the <paramref name="chart" /> where the residuals equal zero. The zero line
+      /// will begin at <paramref name="minObservedDataTime" /> and end at <paramref name="maxObservedDataTime" />
+      /// </summary>
+      DataRepository AddZeroMarkerCurveToChart(AnalysisChartWithLocalRepositories chart, float minObservedDataTime, float maxObservedDataTime);
+
+      /// <summary>
+      ///    Creates the DataRepository for the given residuals <paramref name="outputResidual" /> with given
+      /// <paramref name="id" /> and <paramref name="repositoryName" /> 
+      /// </summary>
+      DataRepository CreateScatterDataRepository(string id, string repositoryName, OutputResiduals outputResidual);
+   }
+
+   public class ResidualsVsTimeChartService : IResidualsVsTimeChartService
+   {
+      private const string ZERO = "Zero";
+      private string _markerCurveId = string.Empty;
+      private readonly IDimensionFactory _dimensionFactory;
+
+      public ResidualsVsTimeChartService(IDimensionFactory dimensionFactory)
+      {
+         _dimensionFactory = dimensionFactory;
+      }
+
+
+      public string Zero => ZERO;
+      public DataRepository AddZeroMarkerCurveToChart(AnalysisChartWithLocalRepositories chart, float minObservedDataTime, float maxObservedDataTime)
+      {
+         var markerRepository = createMarkerRepository(minObservedDataTime, maxObservedDataTime, chart);
+         AddCurvesFor(markerRepository,(column, curve) =>
+         {
+            curve.UpdateMarkerCurve(ZERO);
+            _markerCurveId = curve.Id;
+         }, chart);
+
+         return markerRepository;
+      }
+      private DataRepository createMarkerRepository(float minObservedDataTime, float maxObservedDataTime, AnalysisChartWithLocalRepositories chart)
```

Comment:
Run resharper in this file. It seems to be a bit off

Reply:
done